release/air_tests/air_benchmarks/workloads/gpu_batch_prediction.py

```diff
@@ -0,0 +1,96 @@
+import click
+import time
+import json
+import os
+import numpy as np
+import pandas as pd
+from io import BytesIO
+from typing import List
+
+from PIL import Image
+from torchvision import transforms
+from torchvision.models import resnet18
+
+import ray
+from ray.air.util.tensor_extensions.pandas import TensorArray
+from ray.train.torch import to_air_checkpoint, TorchPredictor
+from ray.train.batch_predictor import BatchPredictor
+from ray.data.preprocessors import BatchMapper
+
+
+def convert_to_pandas(byte_item_list: List[bytes]) -> pd.DataFrame:
+    """
+    Convert input bytes into pandas DataFrame with image column and value of
+    TensorArray to prevent serializing ndarray image data.
+    """
+    images = [
+        Image.open(BytesIO(byte_item)).convert("RGB") for byte_item in byte_item_list
+    ]
+    images = [np.asarray(image) for image in images]
+
+    return pd.DataFrame({"image": TensorArray(images)})
+
+
+def preprocess(df: pd.DataFrame) -> pd.DataFrame:
+    """
+    User Pytorch code to transform user image. Note we still use pandas as
+    intermediate format to hold images as shorthand of python dictionary.
+    """
+    preprocess = transforms.Compose(
+        [
+            transforms.ToTensor(),
+            transforms.Resize(256),
+            transforms.CenterCrop(224),
+            transforms.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225]),
+        ]
+    )
+    df["image"] = df["image"].map(preprocess)
+    df["image"] = df["image"].map(lambda x: x.numpy())
+    df["image"] = TensorArray(df["image"])
+    return df
+
+
+@click.command(help="Run Batch prediction on Pytorch ResNet models.")
+@click.option("--data-size-gb", type=int, default=1)
+def main(data_size_gb: int):
+    data_url = f"s3://air-example-data-2/{data_size_gb}G-image-data-synthetic-raw"
+    print(f"Running GPU batch prediction with {data_size_gb}GB data from {data_url}")
+    start = time.time()
+    dataset = ray.data.read_binary_files(paths=data_url)
+    dataset = dataset.map_batches(convert_to_pandas)
```

Comment:
Add TODO to use ImageFolder once that merges? (https://github.com/ray-project/ray/pull/24641)